services/outputhost/ackmanager.go

```diff
@@ -127,15 +126,21 @@ func (ackMgr *ackManager) getNextAckID(address storeHostAddress, sequence common
 	ackMgr.readLevelSeq++
 
 	if sequence != ackMgr.readLevelSeq {
+		if sequence > ackMgr.readLevelSeq {
+			ackMgr.logger.WithFields(bark.Fields{
+				`old-readLevelSeq`: ackMgr.readLevelSeq,
+				`new-readLevelSeq`: sequence,
+			}).Warn(`adjusting read-level sequence forward`)
 
-		ackMgr.logger.WithFields(bark.Fields{
-			`old-readLevelSeq`: ackMgr.readLevelSeq,
-			`new-readLevelSeq`: sequence,
-		}).Warn(`adjusting read-level sequence`)
-
-		// update gauge here to indicate we skipped messages (potentially due to retention?)
-		ackMgr.cgCache.consumerM3Client.UpdateGauge(metrics.ConsConnectionScope,
-			metrics.OutputhostCGSkippedMessages, int64(sequence-ackMgr.readLevelSeq))
+			// update gauge here to indicate we skipped messages (potentially due to retention?)
+			ackMgr.cgCache.consumerM3Client.UpdateGauge(metrics.ConsConnectionScope,
+				metrics.OutputhostCGSkippedMessages, int64(sequence-ackMgr.readLevelSeq))
+		} else {
+			ackMgr.logger.WithFields(bark.Fields{
+				`old-readLevelSeq`: ackMgr.readLevelSeq,
+				`new-readLevelSeq`: sequence,
+			}).Warn(`adjusting read-level sequence backwards`)
```

Comment:
this should probably be an Error log .. this should really never have to go backwards.